v2/internal/pkg/helm/local_stored_collector.go

```diff
@@ -259,21 +259,21 @@ func repoAdd(chartRepo v2alpha1.Repository) error {
 	if lsc.Downloaders.indexDownloader == nil {
 		indexDownloader, err = helmrepo.NewChartRepository(&entry, getter.All(lsc.Helm.settings))
 		if err != nil {
-			return err
+			return fmt.Errorf("%w %s", err, strings.TrimSpace(chartRepo.Name))
 		}
 	} else {
 		indexDownloader = lsc.Downloaders.indexDownloader
 	}
 
 	if _, err := indexDownloader.DownloadIndexFile(); err != nil {
-		return fmt.Errorf("invalid chart repository %q: %v", chartRepo.URL, err)
+		return fmt.Errorf("invalid chart repository %q: %w", chartRepo.URL, err)
 	}
 
 	// Update temp file with chart entry
 	helmFile.Update(&entry)
 
 	if err := helmFile.WriteFile(lsc.Helm.settings.RepositoryConfig, 0644); err != nil {
-		return fmt.Errorf("error writing helm repo file: %v", err)
+		return fmt.Errorf("error writing helm repo file: %w %s", err, strings.TrimSpace(chartRepo.Name))
```

Comment:
I find it easier to parse if the error message is at the end:
```suggestion
		return fmt.Errorf("error writing helm repo file %s: %w", strings.TrimSpace(chartRepo.Name, err))
```